src/concrete/ml/quantization/quantized_ops.py

```diff
@@ -1577,25 +1577,288 @@ class QuantizedOr(QuantizedOpUnivariateOfEncrypted, QuantizedOp):
+        assert self.output_quant_params.scale is not None
+        assert self.output_quant_params.zero_point is not None
+
+        prepared_inputs = self._prepare_inputs_with_constants(
+            *q_inputs, calibrate=False, quantize_actual_values=True
+        )
+
+        q_input_0: QuantizedArray = prepared_inputs[0]
+        q_input_1: QuantizedArray = prepared_inputs[1]
+
+        assert q_input_0.quantizer.scale is not None
+        assert q_input_0.quantizer.zero_point is not None
+
+        assert q_input_1.quantizer.scale is not None
+        assert q_input_1.quantizer.zero_point is not None
+
+        # Dequantize
+        input_1 = q_input_1.dequant()
+
+        # Replace input_1 with min_non_zero_qvalue if input_1 is 0
+        input_1 = numpy.where(input_1 == 0, self.min_non_zero_value, input_1)
+
+        # Compute the inverse of input_1
+        input_1_inv = 1.0 / input_1
+
+        # Re-quantize the inverse using the same quantization parameters as q_input_1
+        q_input_1_inv_rescaled = self.divider_quantizer.quant(input_1_inv)
+
+        # The product of quantized encrypted integer values
+        product_q_values = q_input_0.qvalues * q_input_1_inv_rescaled
+
+        # mypy
+        assert q_input_0.quantizer.zero_point is not None
+        assert q_input_1.quantizer.zero_point is not None
+        assert self.divider_quantizer.zero_point is not None
+
+        # Integer quantized multiplication need adjustment based on the zero points.
+        if q_input_0.quantizer.zero_point:
+            product_q_values -= q_input_0.quantizer.zero_point * (
+                q_input_1_inv_rescaled - self.divider_quantizer.zero_point
+            )
+        if self.divider_quantizer.zero_point:
+            product_q_values -= self.divider_quantizer.zero_point * (
+                q_input_0.qvalues - q_input_0.quantizer.zero_point
+            )
+
+        # mypy
+        assert self.divider_quantizer.scale is not None
+        assert self.divider_quantizer.zero_point is not None
+
+        # Compute the scale and zero point based on the scale and zero point
+        # of the two quantized values multiplied together
+        new_scale = q_input_0.quantizer.scale * self.divider_quantizer.scale
+        new_zero_point = q_input_0.quantizer.zero_point * self.divider_quantizer.zero_point
+
+        if self.produces_graph_output:
+            return self.make_output_quant_parameters(product_q_values, new_scale, new_zero_point)
+
+        with tag(self.op_instance_name + ".rounding"):
+            # Apply Concrete rounding (if relevant)
+            product_q_values = self.cnp_round(product_q_values, calibrate_rounding)
+
+        # De-quantize the product
+        dequant_product = (product_q_values - new_zero_point) * new_scale
+
+        # Return the raw float values without re-quantizing them to the new scale, as any
+        # following Gemm/Add/Conv will quantize them with _prepare_inputs_with_constants(...)
```

Comment:
but no issue if the div is at a graph's end / if the following ops are like 'reshape' / 'squeeze' ect ?

Reply:
We have that:

```python
        if self.produces_graph_output:
            return self.make_output_quant_parameters(product_q_values, new_scale, new_zero_point)
```

Reply:
ok good to me then !